### PR TITLE
Replace optparse with argparse.

### LINKED ISSUE #10: Port to argparse
We're still using the deprecated optparse. This will be easier after #7

### PR DESCRIPTION
**Pull request by jamadden**
Fixes #10. Currently based on #11 for coverage. Only the last commit here is significant.

### CODE REVIEW

File: src/zope/sendmail/queue.py [hunk 1]
```diff
@@ -362,39 +362,56 @@ class ConsoleApp(object):
...
-                           "the command line." % (INI_SECTION,
-                                                  ', '.join(INI_NAMES)))
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--daemon', action='store_true',
+                        help=("Run in daemon mode, periodically checking queue "
+                              "and sending messages.  Default is to send all "
+                              "messages in queue once and exit."))
+    parser.add_argument('--interval', metavar='<#secs>', type=float, default=3,
+                        help=("How often to check queue when in daemon mode. "
+                              "Default is %(default)s seconds."))
```
Comment (mgedmin) [inline src/zope/sendmail/queue.py]: How did I manage not to know about `%(default)s`???  I was always cursing argparse.ArgumentDefaultsHelpFormatter's all-or-nothing approach.

File: src/zope/sendmail/queue.py [hunk 2]
```diff
@@ -362,39 +362,56 @@ class ConsoleApp(object):
...
+    smtp_group = parser.add_argument_group("SMTP Server",
+                                           "Connection information for the SMTP server")
+    smtp_group.add_argument('--hostname', default='localhost',
+                            help=("Name of smtp host to use for delivery.  Default is "
+                                  "%(default)s."))
+    smtp_group.add_argument('--port', type=int, default=25,
+                            help=("Which port on smtp server to deliver mail to. "
```
Comment (mgedmin) [inline src/zope/sendmail/queue.py]: I would correct the capitalization of SMTP in both of the above help messages.

File: src/zope/sendmail/tests/test_queue.py [hunk 3]
```diff
@@ -423,21 +435,28 @@ def test_args_processing_almost_all_options(self):
...
     def test_args_processing_force_tls_and_no_tls(self):
         # test force_tls and no_tls
         cmdline = "zope-sendmail --force-tls --no-tls %s" % self.dir
-        sys.stderr = self.stderr
-        self.assertRaises(SystemExit, ConsoleApp, cmdline.split(), verbose=False)
+
+        with self.assertRaises(SystemExit):
+            self._make_one(cmdline)
+
+        self.assertIn('--no-tls: not allowed with argument', self._get_output())
```
Comment (mgedmin) [inline src/zope/sendmail/tests/test_queue.py]: That is a weird error message.

File: src/zope/sendmail/queue.py (hunk 1 above)
Comment (jamadden) [reply src/zope/sendmail/queue.py]: I know. I just learned about it too, apparently even optparse supported it.

File: src/zope/sendmail/tests/test_queue.py (hunk 3 above)
Comment (jamadden) [reply src/zope/sendmail/tests/test_queue.py]: That's just what argparse does:
```
$ .tox/py36/bin/zope-sendmail --force-tls --no-tls
usage: zope-sendmail [-h] [--daemon] [--interval <#secs>]
                     [--hostname HOSTNAME] [--port PORT] [--username USERNAME]
                     [--password PASSWORD] [--force-tls | --no-tls]
                     [--config <inifile>]
                     [maildir]
zope-sendmail: error: argument --no-tls: not allowed with argument --force-tls

File: src/zope/sendmail/queue.py (hunk 2 above)
Comment (jamadden) [reply src/zope/sendmail/queue.py]: OK.

**Comment by jamadden** [comment]
Thanks!

**Pull request merged**